demos/dialog/index.js

```diff
@@ -34,3 +34,17 @@ window.toggleActions.onclick = function() {
   dialog.hideActions = hideAction;
   hideActionSpan.innerText = hideAction;
 };
+
+const textField = document.getElementById('dialog8-text-field');
+const primaryButton = document.getElementById('dialog8-primary-action-button');
+    
+primaryButton.addEventListener('click', (e) => {
+  // validate, possible asynchronous such as a server response
```

Comment:
What happens if the validation is asynchronous? You will not be able to stop the propagation of the click event because you'll be waiting for a server response. The click will continue to the next listener and the `dialogAction="close"` will be handled in the dialog's internal listener.

I do not believe you should be decorating the `slot="primaryAction"` button with `dialogAction="close"`. Instead, explicitly encourage users _not_ to do that and manually close the dialog with `dialog.close()`.

That will give users the most flexibility. This approach relies on the validation being synchronous so that you can call `stopPropagation()`. It also relies on the fact that this click listener is firing first and not the click listener within our dialog's implementation.

Reply:
i see!

Reply:
done